dialog/file.go

```diff
@@ -164,75 +182,108 @@ func (f *fileDialog) makeUI() fyne.CanvasObject {
+	f.setView(gridView)
+	f.loadFavorites()
+
+	f.favoritesList = widget.NewList(
+		func() int {
+			return len(f.favorites)
+		},
+		func() fyne.CanvasObject {
+			return container.NewHBox(widget.NewIcon(theme.DocumentIcon()), widget.NewLabel("Template Object"))
+		},
+		func(id widget.ListItemID, item fyne.CanvasObject) {
+			item.(*fyne.Container).Objects[0].(*widget.Icon).SetResource(f.favorites[id].locIcon)
+			item.(*fyne.Container).Objects[1].(*widget.Label).SetText(f.favorites[id].locName)
+		},
+	)
+	f.favoritesList.OnSelected = func(id widget.ListItemID) {
+		f.setLocation(f.favorites[id].loc)
+	}
 
-	favoritesGroup := container.NewVScroll(widget.NewCard("Favorites", "",
-		container.NewVBox(favorites...)))
 	var optionsButton *widget.Button
-	optionsButton = widget.NewButtonWithIcon("Options", theme.SettingsIcon(), func() {
+	optionsButton = widget.NewButtonWithIcon("", theme.SettingsIcon(), func() {
 		f.optionsMenu(fyne.CurrentApp().Driver().AbsolutePositionForObject(optionsButton), optionsButton.Size())
 	})
 
-	left := container.NewBorder(nil, optionsButton, nil, nil, favoritesGroup)
+	var toggleViewButton *widget.Button
+	toggleViewButton = widget.NewButtonWithIcon("", theme.ListIcon(), func() {
+		if f.view == gridView {
+			f.setView(listView)
+			toggleViewButton.SetIcon(theme.GridIcon())
+		} else {
+			f.setView(gridView)
+			toggleViewButton.SetIcon(theme.ListIcon())
+		}
+	})
+
+	optionsbuttons := container.NewHBox(
+		toggleViewButton,
+		optionsButton,
+	)
+
+	header := fyne.NewContainerWithLayout(layout.NewBorderLayout(nil, nil, nil, optionsbuttons),
+		optionsbuttons, widget.NewLabelWithStyle(title, fyne.TextAlignLeading, fyne.TextStyle{Bold: true}))
+
+	footer := fyne.NewContainerWithLayout(layout.NewBorderLayout(nil, nil, nil, buttons),
+		buttons, container.NewHScroll(f.fileName))
 
-	return container.NewBorder(header, footer, left, nil, body)
+	body := container.NewHSplit(f.favoritesList, fyne.NewContainerWithLayout(layout.NewBorderLayout(f.breadcrumbScroll, nil, nil, nil),
+		f.breadcrumbScroll, f.filesScroll))
+	body.SetOffset(0) // Set the minimum offset so that the favoritesList takes only it's minimal width
+
+	return container.NewBorder(header, footer, nil, nil, body)
 }
 
 func (f *fileDialog) optionsMenu(position fyne.Position, buttonSize fyne.Size) {
 	hiddenFiles := widget.NewCheck("Show Hidden Files", func(changed bool) {
 		f.showHidden = changed
 		f.refreshDir(f.dir)
 	})
-	hiddenFiles.SetChecked(f.showHidden)
+	hiddenFiles.Checked = f.showHidden
+	hiddenFiles.Refresh()
 	content := container.NewVBox(hiddenFiles)
 
 	p := position.Add(buttonSize)
-	pos := fyne.NewPos(p.X, p.Y-content.MinSize().Height-theme.Padding()*2)
+	pos := fyne.NewPos(p.X-content.MinSize().Width-theme.Padding()*2, p.Y+theme.Padding()*2)
 	widget.ShowPopUpAtPosition(content, f.win.Canvas, pos)
 }
 
-func (f *fileDialog) loadFavorites() []fyne.CanvasObject {
+func (f *fileDialog) loadFavorites() {
+	f.favorites = []favoriteItem{}
```

Comment:
I think it could be good to move this down to where the `places` variable was used before as it is more clear to how it is being used. As far as I can see, there is no reason that it is set this far up in the method.

Reply:
I think it's better to use the new style
the methods are clearer, loadFavorites is to load the favorites field with the favorites, which can be used later in the favoriteslist
and getPlaces to get the places and return them as a variable (thus the use of get and not load in the method name)
if you are referring to something else, let me know, maybe i didn't get your point exactly

Reply:
No, I just mean that you should move this variable down to just before the for-loop where items are added to it.

Reply:
ah okay, but why it's better to move this down?
the name of method is loadFavorites and it's normal to see the concerned variable get set to an empty slice at first

Reply:
alright, i see your point